pandas/core/indexes/interval.py

```diff
@@ -800,7 +800,7 @@ def astype(self, dtype, copy=True):
     @cache_readonly
     def dtype(self):
         """Return the dtype object of the underlying data"""
-        return IntervalDtype.construct_from_string(str(self.left.dtype))
+        return IntervalDtype(str(self.left.dtype))
```

Comment:
`left.dtype.name`? I'm not sure when these differ, but using `.name` seems safer.